tests_fp/solidity/test_fees_fp.py

```diff
@@ -162,88 +166,92 @@ def test_gov_fee_accrual(amount):
 
 
 @pytest.mark.parametrize("amount", AMOUNT, ids=idfn)
-def test_collect_fees_long(amount):
+def test_collect_fees_long(amount: int):
     """Open a long and then close close to maturity; verify that gov fees are correct"""
-    test = TestFees(trade_amount=amount)  # set up test object
+    test = TestFees(trade_amount=FixedPoint(amount * 10**18))  # set up test object
 
     # check that both gov fees and gov balance are 0 before opening a long
     gov_fees_before_open_long = test.hyperdrive.market_state.gov_fees_accrued * test.hyperdrive.market_state.share_price
-    test.assertEqual(gov_fees_before_open_long, 0)
+    test.assertEqual(int(gov_fees_before_open_long), 0)
     gov_balance_before_open_long = test.gary.wallet.balance.amount
-    test.assertEqual(gov_balance_before_open_long, 0)
+    test.assertEqual(int(gov_balance_before_open_long), 0)
 
     # open long
     test.hyperdrive.open_long(test.bob.wallet, test.trade_amount)
     gov_fees_after_open_long = test.hyperdrive.market_state.gov_fees_accrued * test.hyperdrive.market_state.share_price
     test.assertGreater(gov_fees_after_open_long, gov_fees_before_open_long)
 
     # hyperdrive into the future
-    advance_time(test, 0.5)
+    advance_time(test, FixedPoint("0.5"))
 
     # close long
     test.hyperdrive.close_long(
         agent_wallet=test.bob.wallet,
         bond_amount=test.bob.wallet.longs[0].balance,
-        mint_time=0,
+        mint_time=FixedPoint(0),
     )
 
     # ensure that gov fees have increased
     gov_fees_after_close_long = test.hyperdrive.market_state.gov_fees_accrued * test.hyperdrive.market_state.share_price
     test.assertGreater(gov_fees_after_close_long, gov_fees_after_open_long)
 
     # collect fees to Governance Gary
-    test.gary.wallet.balance.amount += float(
+    test.gary.wallet.balance.amount += (
         test.hyperdrive.market_state.gov_fees_accrued * test.hyperdrive.market_state.share_price
     )
-    test.hyperdrive.market_state.gov_fees_accrued = 0
-    test.assertEqual(test.hyperdrive.market_state.gov_fees_accrued * test.hyperdrive.market_state.share_price, 0)
+    # TODO: fix this check, don't set fees accrued to zero
+    test.hyperdrive.market_state.gov_fees_accrued = FixedPoint(0)
+    test.assertEqual(int(test.hyperdrive.market_state.gov_fees_accrued * test.hyperdrive.market_state.share_price), 0)
 
     gov_balance_after = test.gary.wallet.balance.amount
     # ensure that Governance Gary's balance has increased
     test.assertGreater(gov_balance_after, gov_balance_before_open_long)
-    # ensure that Governance Gary got the exaxt fees expected
-    test.assertAlmostEqual(gov_balance_after, gov_fees_after_close_long, delta=1e-16 * test.trade_amount)
+    # ensure that Governance Gary got the exaxt fees FixedPointexpecteFixedPointd
+    test.assertAlmostEqual(
+        float(gov_balance_after), float(gov_fees_after_close_long), delta=1e-16 * float(test.trade_amount)
```

Comment:
I think this should probably just use `places` now that we're in FixedPoint world. We always expect accuracy up to 18 decimals, regardless of scale, and never guarantee more.

Reply:
this goes for everywhere else I think

Reply:
places doesn't work well with float since large values will have less place precision.  delta would be better if we used ints here actually, maybe should change to that...